lib-core/src/factories/AbstractFactory.ts

```diff
@@ -1,12 +1,18 @@
-import { BaseModel } from "../models/BaseModel";
+import { BaseModel } from '../models/BaseModel';
+import { DiagramEngine } from '../DiagramEngine';
 
 export abstract class AbstractFactory<T extends BaseModel> {
 	type: string;
+	protected engine: DiagramEngine;
 
 	constructor(name: string) {
 		this.type = name;
 	}
 
+	setDiagramEngine(engine: DiagramEngine) {
+		this.engine = engine;
+	}
+
 	getType(): string {
 		return this.type;
 	}
```

Comment:
Also, if there is type getter, maybe it's better to make type property as private or protected?